src/content/2.4/code/agda/snippet15.agda

```diff
@@ -0,0 +1,5 @@
+instance
+  repStream : Representable Stream
+  repStream .Repf = ℕ
+  repStream .tabulate f = Cons (f 0) (tabulate (f ∘ (_+ 1)))
+  repStream .index (Cons b bs) n = if (n == 0) then b else index bs (n - 1)
```

Comment:
Now I am wondering 🤔  if Representable functors are abstraction behind model based testing. Perhaps this is interesting topic to dig deeper at some point.